### ISSUE TITLE
Should not lint on curried functions

### ISSUE THREAD

**Issue by radbasa** [Appsilon/box.linters#16]
## box.linters version

0.0.0.9006

## Source code to lint

```R
some_function <- function(x) {
  function(y) {
    x + y
  }
}

this_fun <- some_function(1)
this_fun(2)
```

## Lint command

```R
lintr::lint(code, linters = lintr::linters_with_defaults(defaults = box.linters::rhino_default_linters))
```

## Lint result

```
<text>:9:1: warning: [box_usage_linter] Function not imported nor defined.
this_fun(2)
^~~~~~~~
```

## Expected result

Should not lint on curried functions.
